packages/slo-stat-bot/src/slo-logic.ts

```diff
@@ -0,0 +1,504 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {GitHubAPI} from 'probot/lib/github';
+import {Context} from 'probot';
+import moment from 'moment';
+
+interface SLOStatus {
+  appliesTo: boolean;
+  isCompliant: boolean | null;
+}
+
+enum Priority {
+  'P0',
+  'P1',
+  'P2',
+  'P3',
+  'P4',
+  'P5',
+}
+
+interface SLORules {
+  appliesTo: {
+    gitHubLabels?: string | string[];
+    excludedGitHubLabels?: string | string[];
+    priority?: Priority;
+    issueType?: string;
+    issues: boolean;
+    prs: boolean;
+  };
+  complianceSettings: {
+    responseTime: string | number;
+    resolutionTime: string | number;
+    requiresAssignee: boolean;
+    responders?: {
+      owners?: string | string[];
+      contributors?: string;
+      users?: string[];
+    };
+  };
+}
+
+interface IssueAssignees {
+  login: string;
+}
+
+interface IssuesListCommentsItem {
+  id: number;
+  user: {
+    login: string;
+  };
+  created_at: string;
+  updated_at: string;
+}
+
+interface ReposListCollaboratorsItem {
+  login: string;
+  permissions: {
+    pull: boolean;
+    push: boolean;
+    admin: boolean;
+  };
+}
+
+//Checking if slo applies to a given issue
+getSLOStatus.appliesTo = async function appliesTo(
```

Comment:
Super nitpicky, can we have a slightly more descriptive title for this function?